packages/typescript-estree/src/node-utils.ts

```diff
@@ -223,7 +223,10 @@ export function canContainDirective(
  * @param ast the AST object
  * @returns the range data
  */
-export function getRange(node: ts.Node, ast: ts.SourceFile): [number, number] {
+export function getRange(
+  node: Pick<ts.Node, 'getEnd' | 'getStart'>,
+  ast: ts.SourceFile,
+): [number, number] {
```

Comment:
`Program`-typed values were getting passed in here. Having the `node` `Pick` out the only methods it needs stopped some type errors.